src/store/modules/recipes/__tests__/actions.test.ts

```diff
@@ -0,0 +1,270 @@
+      expect(commit).toHaveBeenCalledTimes(3);
+    });
+  });
+
+  describe("ADD_RECIPE", () => {
+    let commit: typeof jest.fn;
+
+    const addRecipe = actions[ADD_RECIPE] as ActionHandler<State, RootState>;
+
+    const newRecipe = {
+      title: "Some title",
+      category: "Some category",
+      instructions: "1. Step 1; 2. Step 2",
+    };
+
+    beforeEach(() => {
+      commit = jest.fn();
+    });
+
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it("commits add recipe mutation if there was no error", async () => {
+      await addRecipe.call(
+        storeContext,
+        { commit } as unknown as ActionContext<State, RootState>,
+        newRecipe
+      );
+
+      expect(commit).toHaveBeenCalledWith(
+        Mutations.SET_DATA_STATUS,
+        DataStatus.PENDING
+      );
+      expect(commit).toHaveBeenCalledWith(Mutations.ADD_RECIPE, newRecipe);
+      expect(commit).toHaveBeenCalledWith(
+        Mutations.SET_DATA_STATUS,
+        DataStatus.FULFILLED
+      );
+      expect(commit).toHaveBeenCalledTimes(3);
+    });
+
+    it("commits set error mutation if error is thrown", async () => {
+      const error = new Error("Some Error");
+
+      jest.spyOn(global, "setTimeout").mockImplementationOnce(() => {
+        throw error;
+      });
+
+      await addRecipe.call(
+        storeContext,
+        { commit } as unknown as ActionContext<State, RootState>,
+        newRecipe
+      );
+
+      expect(commit).toHaveBeenCalledWith(
+        Mutations.SET_DATA_STATUS,
+        DataStatus.PENDING
+      );
+      expect(commit).toHaveBeenCalledWith(Mutations.SET_ERROR, error);
+      expect(commit).toHaveBeenCalledWith(
+        Mutations.SET_DATA_STATUS,
+        DataStatus.REJECTED
+      );
+      expect(commit).toHaveBeenCalledTimes(3);
+    });
+  });
+
+  describe("UPDATE_RECIPE", () => {
+    let commit: typeof jest.fn;
+
+    const updateRecipe = actions[UPDATE_RECIPE] as ActionHandler<
+      State,
+      RootState
+    >;
+
+    const updatedRecipe = {
+      title: "Some title",
+      category: "Some category",
+      instructions: "1. Step 1; 2. Step 2",
+    };
+
+    beforeEach(() => {
+      commit = jest.fn();
+    });
+
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it("commits update recipe mutation if there was no error", async () => {
+      await updateRecipe.call(
+        storeContext,
+        { commit } as unknown as ActionContext<State, RootState>,
+        updatedRecipe
+      );
+
+      expect(commit).toHaveBeenCalledWith(
+        Mutations.SET_DATA_STATUS,
+        DataStatus.PENDING
+      );
+      expect(commit).toHaveBeenCalledWith(
+        Mutations.UPDATE_RECIPE,
+        updatedRecipe
+      );
+      expect(commit).toHaveBeenCalledWith(
+        Mutations.SET_DATA_STATUS,
+        DataStatus.FULFILLED
+      );
+      expect(commit).toHaveBeenCalledTimes(3);
+    });
+
+    it("commits set error mutation if error is thrown", async () => {
```

Comment:
```suggestion
    it("should not update recipe if error is thrown", async () => {
```

Reply:
you also should check if `Mutations.UPDATE_RECIPE` not called. You should ensure that actions are not called if an error occurs